tests/AbstractRepositoryTest.php

```diff
@@ -718,4 +774,100 @@ public function testFindExpectsStringPath()
 
         $repo->find(new \stdClass());
     }
+
+    public function testLinkDirectory()
+    {
+        $repo = $this->createAutoresolveLinksRepository();
```

Comment:
I'm not a fan of this creation method. The method largely hides what the test is actually testing. If I read the test, I wonder why the assertions below should hold, and the reason lies in the fixture setup (which link points to where, ...). I think you should inline this method everywhere and reduce it to the minimum required for each test.